scout/models/event.py

```diff
@@ -26,34 +29,78 @@
 Variant:
   - Comments
   - Sanger sequencing orders
-  - Pinning?
-  - Archivals?
+  - Pinning
+  - Archivals
+
 """
+
 from __future__ import absolute_import, unicode_literals
 from datetime import datetime
 
-from mongoengine import (DateTimeField, EmbeddedDocument, ListField,
+from mongoengine import (DateTimeField, Document, ListField,
                          ReferenceField, StringField)
 
+VERBS = (
+  "assign",
+  "unassign",
+  "status",
+  "comment",
+  "synopsis",
+  "pin",
+  "unpin",
+  "sanger",
+  "archive",
+  "open_research",
+)
 
-class Event(EmbeddedDocument):
+class Event(Document):
   """Embedded model for defining a general user generated event."""
-  title = StringField()
-  content = StringField()
+  # an event will allways belong to a institute and a case
+  institute = ReferenceField('Institute', required=True)
+  case_id = StringField(required=True)
+  # All events will have url links
   link = StringField()
+  # All events has to have a category
+  category = StringField(choices=('case', 'variant'), required=True)
+                             
+  # All events will have an author
+  author = ReferenceField('User', required=True)
+  # Subject is the string that will be displayed after 'display_info'
+  subject = StringField(required=True) # case 23 or 1_2343_A_C
+  
+  verb = StringField(choices=VERBS)
+  level = StringField(choices=('global', 'specific'), default='specific')
 
-  # metadata
-  author = ReferenceField('User')      # George
-  verb = StringField()                 # commented on
-  subject = StringField()              # case 23
-  action = StringField()
-  tags = ListField(StringField())
-  institute = ReferenceField('Institute')
-
+  # An event can belong to a variant
+  variant_id = StringField()
+  # This is the content of a comment
+  content = StringField()
+  
   # timestamps
   created_at = DateTimeField(default=datetime.now)
   updated_at = DateTimeField(default=datetime.now)
 
+  def get_display_info(self):
+    """
+    Return the string that should be displayed based on the keyword
+    """
+    display_info = {
+      "assign" : "was assigned to",
+      "unassign" : "was unassigned from",
+      "status" : "updated the status for",
+      "comment" : "commented on",
+      "synopsis" : "updated synopsis for",
+      "pin" : "pinned variant",
+      "unpin" : "removed pinned variant",
+      "sanger" : "ordered sanger sequencing for",
+      "archive" : "archived",
+      "open_research" : "opened research mode for",
+    }
+    
+    return display_info.get(self.verb, "")
```

Comment:
Any reason for returning an empty string by default? I was thinking it's more robust to test for ``None``